### PR TITLE
Add compiler version

### PR DESCRIPTION
**Pull request by AnotherButler**
Add ARMC6 version in the Online Compiler.

### CODE REVIEW

**Comment by MarceloSalazar** [comment]
@AnotherButler 
Thanks for the PR! I think the doc is missing a bit of context. 
The default version of the toolchain in the Online Compiler is still Arm Compiler 5.03 Update 3 (same as listed above).
However, for v8-M platforms only, I understand we've introduced support for Arm Compiler 6.
I don't think we have a way to let developers chose the compiler version in the Online Compiler.

**Comment by theotherjimmy** [comment]
@MarceloSalazar We use Arm Compiler 5.06 in the online compiler. 

@AnotherButler The online compiler switches the backing toolchain based on architecture version as follows:

| Arch Version | Compiler |
|----                  |----           |
| Arm v6M       | Arm Compiler 5.06 |
| Arm v7M       | Arm Compiler 5.06 |
| Arm v7A        | Arm Compiler 5.06 |
| Arm v8M       | Arm Compiler 6.10.1 |

File: docs/tools/tools_intro.md [hunk 1]
```diff
@@ -27,9 +27,16 @@ Mbed OS 5 can be built with various toolchains. The currently supported versions
...
 ##### Arm Mbed Online Compiler
 
-The Mbed Online Compiler is our in-house IDE, and should be familiar to anyone who's been working with Mbed for a while. For more information, see the [Online Compiler page](developing-mbed-online-compiler.html).
+The Mbed Online Compiler is our in-house IDE, and should be familiar to anyone who's been working with Mbed for a while. It switches the backing toolchain based on the architecture version:
 
-For reference, the ARMC6 version the Online Compiler uses is 6.10.
+| Architecture version | Compiler |
+|----                  |----           |
+| Arm v6M       | Arm Compiler 5.06 |
+| Arm v7M       | Arm Compiler 5.06 |
```
Comment (MarceloSalazar) [inline docs/tools/tools_intro.md]: Can we please add full details on the compile version?
`Arm Compiler 5.06 Update 3`

Comment (AnotherButler) [reply docs/tools/tools_intro.md]: If I recall correctly, there was a Product decision earlier not to include the update version. Would you like us to revisit the topic?

Comment (MarceloSalazar) [reply docs/tools/tools_intro.md]: Yes please, let's discuss over email, can you loop me in please?
IMO we need to facilitate reproducibility of defects. The way to achieve this is by being transparent on dependencies, including tools. Thanks!

Comment (AnotherButler) [reply docs/tools/tools_intro.md]: I started typing this email but realized we list the update number above (and have for the past five releases), so I must be thinking of something else. I'm fine including the update numbers here.

**Comment by MarceloSalazar** [comment]
Thanks @AnotherButler !

**Pull request merged**